actors/builtin/market/market_actor.go

```diff
@@ -381,6 +382,11 @@ func (a Actor) OnMinerSectorsTerminate(rt Runtime, params *OnMinerSectorsTermina
 
 		for _, dealID := range params.DealIDs {
 			deal, err := proposals.Get(dealID)
+			// deal could have terminated and hence deleted before the sector is terminated.
+			// we should simply continue instead of aborting execution here if a deal is not found.
+			if errors.Unwrap(err) == errDealNotFound {
```

Comment:
Gah, not your fault but this is the breaking straw for me. Please refactor `proposals.Get` to return three values `(deal, found bool, err error)`, distinguishing these cases without introducing error introspection patterns.

Reply:
This is done.